internal/access/access_key.go

```diff
@@ -57,9 +57,7 @@ func CreateAccessKey(c *gin.Context, accessKey *models.AccessKey) (body string,
 	return body, err
 }
 
-func DeleteAccessKey(c *gin.Context, id uid.ID, name string) error {
-	rCtx := GetRequestContext(c)
-
+func DeleteAccessKey(rCtx RequestContext, id uid.ID, name string) error {
```

Comment:
I changed from `gin.Context` to `RequestContext` here because the tests in `access/access_key_test` were failing due to this change. By accepting the `RequestContext` I can avoid mutating the `gin.Context`, which was being re-used by subsequent test cases.

I believe we have a few other access functions that already accept `RequestContext` directly, and long term I'd like to see us use `RequestContext` directly in both the API handlers and the access functions.